app/models/topic_hot_score.rb

```diff
@@ -5,6 +5,22 @@ class TopicHotScore < ActiveRecord::Base
 
   DEFAULT_BATCH_SIZE = 1000
 
+  def self.hot_topic_ids_cache
+    @hot_topic_ids_cache ||= DistributedCache.new("hot_topic_ids_cache")
+  end
+
+  def self.hottest_topic_ids
+    hot_topic_ids_cache["ids"].to_a
+  end
+
+  def self.recreate_hottest_topic_ids_cache
+    hot_topic_ids_cache.clear
+
+    hot_topic_ids_cache.defer_get_set("ids") do
+      DB.query_single("SELECT topic_id FROM topic_hot_scores ORDER BY score DESC LIMIT 100")
```

Comment:
We can just store this as a set instead of an array to make lookups faster O(1) complexity.